cla_public/libs/laalaa.py

```diff
@@ -54,10 +54,11 @@ def decode_categories(result):
     return result
 
 
-def find(postcode, category=None, page=1):
-
-    data = laalaa_search(postcode=postcode, category=category, page=page)
-
-    data["results"] = map(decode_categories, data.get("results", []))
-
-    return data
+def find(postcode, categories=None, page=1):
+    merged_data = {"results": []}
+    for category in categories:
+        data = laalaa_search(postcode=postcode, category=category, page=page)
+        data["results"] = map(decode_categories, data.get("results", []))
+        merged_data["results"].extend(data["results"])
```

Comment:
As we're just extending the results into a new dict's `results` (rather than returning the original dict, modified), there doesn't seem any point doing this in two steps (putting the mapped results into the dict, then getting them from there to extend the merged dict)

Reply:
Less ugly than I thought it would be.

Reply:
Ah it was aesthetics. Fair. I figured it was just a hangover from the previous pattern :)